From 63e2e3e197c80941274cbbaf66de1839846e116c Mon Sep 17 00:00:00 2001
From: Gilles Gouaillardet <gilles.gouaillardet@iferc.org>
Date: Wed, 29 Jan 2014 14:27:29 +0900
Subject: [PATCH 3/3] Fix prototypes and remove unused local variables. See #2010

Use type (const void *) instead of (void *) and explicitly cast
to (void *) when needed in order to makes gcc -Wall happy.
---
 src/mpi/romio/adio/ad_lustre/ad_lustre.h          |    6 ++--
 src/mpi/romio/adio/ad_lustre/ad_lustre_hints.c    |    1 -
 src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c |   34 ++++++++++----------
 src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c   |   14 ++++----
 src/mpi/romio/adio/ad_lustre/ad_lustre_wrstr.c    |    2 +-
 5 files changed, 28 insertions(+), 29 deletions(-)

diff --git a/src/mpi/romio/adio/ad_lustre/ad_lustre.h b/src/mpi/romio/adio/ad_lustre/ad_lustre.h
index ae572cc..93034cd 100644
--- a/src/mpi/romio/adio/ad_lustre/ad_lustre.h
+++ b/src/mpi/romio/adio/ad_lustre/ad_lustre.h
@@ -48,15 +48,15 @@ void ADIOI_LUSTRE_ReadContig(ADIO_File fd, void *buf, int count,
                              MPI_Datatype datatype, int file_ptr_type,
                              ADIO_Offset offset, ADIO_Status *status,
                              int *error_code);
-void ADIOI_LUSTRE_WriteContig(ADIO_File fd, void *buf, int count,
+void ADIOI_LUSTRE_WriteContig(ADIO_File fd, const void *buf, int count,
                               MPI_Datatype datatype, int file_ptr_type,
                               ADIO_Offset offset, ADIO_Status *status,
                               int *error_code);
-void ADIOI_LUSTRE_WriteStrided(ADIO_File fd, void *buf, int count,
+void ADIOI_LUSTRE_WriteStrided(ADIO_File fd, const void *buf, int count,
 			       MPI_Datatype datatype, int file_ptr_type,
 			       ADIO_Offset offset, ADIO_Status *status,
 			       int *error_code);
-void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, void *buf, int count,
+void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, const void *buf, int count,
 		                   MPI_Datatype datatype, int file_ptr_type,
 		                   ADIO_Offset offset, ADIO_Status *status,
                                    int *error_code);
diff --git a/src/mpi/romio/adio/ad_lustre/ad_lustre_hints.c b/src/mpi/romio/adio/ad_lustre/ad_lustre_hints.c
index aaab557..7b70ba3 100644
--- a/src/mpi/romio/adio/ad_lustre/ad_lustre_hints.c
+++ b/src/mpi/romio/adio/ad_lustre/ad_lustre_hints.c
@@ -18,7 +18,6 @@ void ADIOI_LUSTRE_SetInfo(ADIO_File fd, MPI_Info users_info, int *error_code)
     int flag, stripe_val[3], str_factor = -1, str_unit=0, start_iodev=-1;
     struct lov_user_md lum = { 0 };
     int err, myrank, fd_sys, perm, amode, old_mask;
-    int int_val, tmp_val;
     static char myname[] = "ADIOI_LUSTRE_SETINFO";
 
     value = (char *) ADIOI_Malloc((MPI_MAX_INFO_VAL+1)*sizeof(char));
diff --git a/src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c b/src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c
index 6869f58..3dfdb48 100644
--- a/src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c
+++ b/src/mpi/romio/adio/ad_lustre/ad_lustre_rwcontig.c
@@ -14,9 +14,9 @@
 
 #define LUSTRE_MEMALIGN (1<<12)  /* to use page_shift */
 
-static void ADIOI_LUSTRE_Aligned_Mem_File_Write(ADIO_File fd, void *buf, int len, 
+static void ADIOI_LUSTRE_Aligned_Mem_File_Write(ADIO_File fd, const void *buf, int len,
               ADIO_Offset offset, int *err);
-static void ADIOI_LUSTRE_Aligned_Mem_File_Write(ADIO_File fd, void *buf, int len, 
+static void ADIOI_LUSTRE_Aligned_Mem_File_Write(ADIO_File fd, const void *buf, int len,
               ADIO_Offset offset, int *err)
 {
     int rem, size, nbytes;
@@ -33,29 +33,29 @@ static void ADIOI_LUSTRE_Aligned_Mem_File_Write(ADIO_File fd, void *buf, int len
     }
 }
 
-static void ADIOI_LUSTRE_Aligned_Mem_File_Read(ADIO_File fd, void *buf, int len, 
+static void ADIOI_LUSTRE_Aligned_Mem_File_Read(ADIO_File fd, const void *buf, int len,
               ADIO_Offset offset, int *err);
-static void ADIOI_LUSTRE_Aligned_Mem_File_Read(ADIO_File fd, void *buf, int len, 
+static void ADIOI_LUSTRE_Aligned_Mem_File_Read(ADIO_File fd, const void *buf, int len,
               ADIO_Offset offset, int *err)
 {
     int rem, size, nbytes;
     if (!(len % fd->d_miniosz) && (len >= fd->d_miniosz))
-	*err = pread(fd->fd_direct, buf, len, offset);
+	*err = pread(fd->fd_direct, (void *)buf, len, offset);
     else if (len < fd->d_miniosz)
-	*err = pread(fd->fd_sys, buf, len, offset);
+	*err = pread(fd->fd_sys, (void *)buf, len, offset);
     else {
 	rem = len % fd->d_miniosz;
 	size = len - rem;
-	nbytes = pread(fd->fd_direct, buf, size, offset);
+	nbytes = pread(fd->fd_direct, (void *)buf, size, offset);
 	nbytes += pread(fd->fd_sys, ((char *)buf) + size, rem, offset+size);
 	*err = nbytes;
     }
 }
 
 
-static int ADIOI_LUSTRE_Directio(ADIO_File fd, void *buf, int len, 
+static int ADIOI_LUSTRE_Directio(ADIO_File fd, const void *buf, int len,
 			   off_t offset, int rw);
-static int ADIOI_LUSTRE_Directio(ADIO_File fd, void *buf, int len, 
+static int ADIOI_LUSTRE_Directio(ADIO_File fd, const void *buf, int len,
 			   off_t offset, int rw)
 {
     int err=-1, diff, size=len, nbytes = 0;
@@ -65,9 +65,9 @@ static int ADIOI_LUSTRE_Directio(ADIO_File fd, void *buf, int len,
 	diff = fd->d_miniosz - (offset % fd->d_miniosz);
 	diff = ADIOI_MIN(diff, len);
 	if (rw)
-	    nbytes = pwrite(fd->fd_sys, buf, diff, offset);
+	    nbytes = pwrite(fd->fd_sys, (void *)buf, diff, offset);
 	else
-	    nbytes = pread(fd->fd_sys, buf, diff, offset);
+	    nbytes = pread(fd->fd_sys, (void *)buf, diff, offset);
 	buf = ((char *) buf) + diff;
 	offset += diff;
 	size = len - diff;
@@ -100,22 +100,22 @@ static int ADIOI_LUSTRE_Directio(ADIO_File fd, void *buf, int len,
 	    newbuf = (void *) memalign(LUSTRE_MEMALIGN, size);
 	    if (newbuf) {
 		ADIOI_LUSTRE_Aligned_Mem_File_Read(fd, newbuf, size, offset, &err);
-		if (err > 0) memcpy(buf, newbuf, err);
+		if (err > 0) memcpy((void *)buf, newbuf, err);
 		nbytes += err;
 		ADIOI_Free(newbuf);
 	    }
-	    else nbytes += pread(fd->fd_sys, buf, size, offset);
+	    else nbytes += pread(fd->fd_sys, (void *)buf, size, offset);
 	}
 	err = nbytes;
     }
     return err;
 }
 
-static void ADIOI_LUSTRE_IOContig(ADIO_File fd, void *buf, int count, 
+static void ADIOI_LUSTRE_IOContig(ADIO_File fd, const void *buf, int count,
                    MPI_Datatype datatype, int file_ptr_type,
 	           ADIO_Offset offset, ADIO_Status *status, 
 		   int io_mode, int *error_code);
-static void ADIOI_LUSTRE_IOContig(ADIO_File fd, void *buf, int count, 
+static void ADIOI_LUSTRE_IOContig(ADIO_File fd, const void *buf, int count,
                    MPI_Datatype datatype, int file_ptr_type,
 	           ADIO_Offset offset, ADIO_Status *status, 
 		   int io_mode, int *error_code)
@@ -149,7 +149,7 @@ static void ADIOI_LUSTRE_IOContig(ADIO_File fd, void *buf, int count,
 #ifdef ADIOI_MPE_LOGGING
         MPE_Log_event(ADIOI_MPE_read_a, 0, NULL);
 #endif
-	    err = read(fd->fd_sys, buf, len);
+	    err = read(fd->fd_sys, (void *)buf, len);
 #ifdef ADIOI_MPE_LOGGING
         MPE_Log_event(ADIOI_MPE_read_b, 0, NULL);
 #endif
@@ -184,7 +184,7 @@ ioerr:
     /* --END ERROR HANDLING-- */
 }
 
-void ADIOI_LUSTRE_WriteContig(ADIO_File fd, void *buf, int count, 
+void ADIOI_LUSTRE_WriteContig(ADIO_File fd, const void *buf, int count,
                    MPI_Datatype datatype, int file_ptr_type,
 	           ADIO_Offset offset, ADIO_Status *status, int *error_code)
 {
diff --git a/src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c b/src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c
index 89c0999..58d75c9 100644
--- a/src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c
+++ b/src/mpi/romio/adio/ad_lustre/ad_lustre_wrcoll.c
@@ -12,7 +12,7 @@
 #include "adio_extern.h"
 
 /* prototypes of functions used for collective writes only. */
-static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, void *buf,
+static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, const void *buf,
 					MPI_Datatype datatype, int nprocs,
 					int myrank,
 					ADIOI_Access *others_req,
@@ -22,7 +22,7 @@ static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, void *buf,
 					int contig_access_count,
 					int *striping_info,
                                         int **buf_idx, int *error_code);
-static void ADIOI_LUSTRE_Fill_send_buffer(ADIO_File fd, void *buf,
+static void ADIOI_LUSTRE_Fill_send_buffer(ADIO_File fd, const void *buf,
 					  ADIOI_Flatlist_node *flat_buf,
 					  char **send_buf,
 					  ADIO_Offset *offset_list,
@@ -35,7 +35,7 @@ static void ADIOI_LUSTRE_Fill_send_buffer(ADIO_File fd, void *buf,
                                           int *curr_to_proc,
 					  int *done_to_proc, int iter,
 					  MPI_Aint buftype_extent);
-static void ADIOI_LUSTRE_W_Exchange_data(ADIO_File fd, void *buf,
+static void ADIOI_LUSTRE_W_Exchange_data(ADIO_File fd, const void *buf,
 					 char *write_buf,
 					 ADIOI_Flatlist_node *flat_buf,
 					 ADIO_Offset *offset_list,
@@ -59,7 +59,7 @@ void ADIOI_Heap_merge(ADIOI_Access *others_req, int *count,
                       ADIO_Offset *srt_off, int *srt_len, int *start_pos,
                       int nprocs, int nprocs_recv, int total_elements);
 
-void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, void *buf, int count,
+void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, const void *buf, int count,
 				   MPI_Datatype datatype,
 				   int file_ptr_type, ADIO_Offset offset,
 				   ADIO_Status *status, int *error_code)
@@ -283,7 +283,7 @@ void ADIOI_LUSTRE_WriteStridedColl(ADIO_File fd, void *buf, int count,
 /* If successful, error_code is set to MPI_SUCCESS.  Otherwise an error
  * code is created and returned in error_code.
  */
-static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, void *buf,
+static void ADIOI_LUSTRE_Exch_and_write(ADIO_File fd, const void *buf,
 					MPI_Datatype datatype, int nprocs,
 					int myrank, ADIOI_Access *others_req,
                                         ADIOI_Access *my_req,
@@ -613,7 +613,7 @@ over:
 /* Sets error_code to MPI_SUCCESS if successful, or creates an error code
  * in the case of error.
  */
-static void ADIOI_LUSTRE_W_Exchange_data(ADIO_File fd, void *buf,
+static void ADIOI_LUSTRE_W_Exchange_data(ADIO_File fd, const void *buf,
 					 char *write_buf,
 					 ADIOI_Flatlist_node *flat_buf,
 					 ADIO_Offset *offset_list,
@@ -885,7 +885,7 @@ static void ADIOI_LUSTRE_W_Exchange_data(ADIO_File fd, void *buf,
     ADIOI_BUF_INCR \
 }
 
-static void ADIOI_LUSTRE_Fill_send_buffer(ADIO_File fd, void *buf,
+static void ADIOI_LUSTRE_Fill_send_buffer(ADIO_File fd, const void *buf,
 					  ADIOI_Flatlist_node *flat_buf,
 					  char **send_buf,
 					  ADIO_Offset *offset_list,
diff --git a/src/mpi/romio/adio/ad_lustre/ad_lustre_wrstr.c b/src/mpi/romio/adio/ad_lustre/ad_lustre_wrstr.c
index ea140ef..2f1ffb9 100644
--- a/src/mpi/romio/adio/ad_lustre/ad_lustre_wrstr.c
+++ b/src/mpi/romio/adio/ad_lustre/ad_lustre_wrstr.c
@@ -144,7 +144,7 @@
     } \
 }
 
-void ADIOI_LUSTRE_WriteStrided(ADIO_File fd, void *buf, int count,
+void ADIOI_LUSTRE_WriteStrided(ADIO_File fd, const void *buf, int count,
 			       MPI_Datatype datatype, int file_ptr_type,
 			       ADIO_Offset offset, ADIO_Status * status,
 			       int *error_code)
-- 
1.7.1

